File: fixtures/scale/java/large/src/cryptobench/ciphers/affine/AffineSuite.java
package cryptobench.ciphers.affine;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class AffineSuite implements CipherSuite {
    @Override
    public String name() {
        return "affine";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new AffineCipher(AffineKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : AffineVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/affine/AffineSuite.kt
package cryptobench.ciphers.affine

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class AffineSuite : CipherSuite {
    override fun name(): String = "affine"

    override fun run(): SuiteResult {
        val cipher: Cipher = AffineCipher(AffineKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in AffineVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
